### PR TITLE
feat(e2e_simulator.launch): add argument for running the CARLA interface

### PR DESCRIPTION
**Pull request by mraditya01**
## Description

Added additional arguments on e2e_launch for running Autoware with CARLA.
Needed to use the `carla_autoware` package to integrate Autoware with the CARLA simulator. See https://github.com/autowarefoundation/autoware.universe/pull/6859

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_launch/launch/e2e_simulator.launch.xml [hunk 1]
```diff
@@ -6,6 +6,7 @@
   <arg name="vehicle_model" default="sample_vehicle" description="vehicle model name"/>
   <arg name="sensor_model" default="sample_sensor_kit" description="sensor model name"/>
   <arg name="data_path" default="$(env HOME)/autoware_data" description="packages data and artifacts directory path"/>
+  <arg name="carla" default="false" description="simulation used"/>
```
Comment (maxime-clem) [inline autoware_launch/launch/e2e_simulator.launch.xml:9]:
```suggestion
  <arg name="carla" default="false" description="also launch the interface with the CARLA simulator"/>
```

**Review by maxime-clem — CHANGES_REQUESTED** [review]

File: autoware_launch/launch/e2e_simulator.launch.xml [hunk 2]
```diff
@@ -6,6 +6,7 @@
   <arg name="vehicle_model" default="sample_vehicle" description="vehicle model name"/>
   <arg name="sensor_model" default="sample_sensor_kit" description="sensor model name"/>
   <arg name="data_path" default="$(env HOME)/autoware_data" description="packages data and artifacts directory path"/>
+  <arg name="carla" default="false" description="also launch the interface with the CARLA simulator"/>
```
Comment (maxime-clem) [inline autoware_launch/launch/e2e_simulator.launch.xml:9]: Let's change the argument to be `simulator_type`.
```suggestion
  <arg name="simulator_type" default="awsim" description="select the simulator type ('awsim' or 'carla')"/>
```

Comment (mraditya01) [reply autoware_launch/launch/e2e_simulator.launch.xml:9]: Regarding this, XML Launch file cannot accept string as arguments of `if` on line 42. I will try to find the work around to make it work.

Comment (mraditya01) [reply autoware_launch/launch/e2e_simulator.launch.xml]: Fixed, changed the arguments and the conditional launch to `eval`.

**Review by maxime-clem — APPROVED** [review]

File: autoware_launch/launch/e2e_simulator.launch.xml [hunk 3]
```diff
@@ -31,15 +33,30 @@
...
-  />
+
   <!-- Tools -->
   <arg name="rviz" default="true" description="launch rviz"/>
   <arg name="rviz_config" default="$(find-pkg-share autoware_launch)/rviz/autoware.rviz" description="rviz config"/>
 
+  <group scoped="false" if="$(eval '\'$(var simulator_type)\' == \'carla\'')">
+    <include file="$(find-pkg-share carla_autoware)/carla_autoware.launch.xml">
+      <arg name="carla_map_name" value="$(var carla_map)"/>
+    </include>
```
Comment (yukkysaito) [inline autoware_launch/launch/e2e_simulator.launch.xml:44]: What is the purpose of the file carla_autoware.launch.xml?
What is the related PR if necessary?

**Review by yukkysaito — CHANGES_REQUESTED** [review]
Thanks for the PR. I made some comments.

File: autoware_launch/launch/e2e_simulator.launch.xml (hunk 3 above)
Comment (maxime-clem) [reply autoware_launch/launch/e2e_simulator.launch.xml:44]: `carla_autoware.launch.xml` will launch the nodes to communicate between CARLA and Autoware.

File: autoware_launch/launch/e2e_simulator.launch.xml [hunk 4]
```diff
@@ -6,6 +6,8 @@
   <arg name="vehicle_model" default="sample_vehicle" description="vehicle model name"/>
   <arg name="sensor_model" default="sample_sensor_kit" description="sensor model name"/>
   <arg name="data_path" default="$(env HOME)/autoware_data" description="packages data and artifacts directory path"/>
+  <arg name="simulator_type" default="awsim" description="select the simulator type ('awsim' or 'carla')"/>
+  <arg name="carla_map" default="Town01" description="Map used for CARLA"/>
```
[the remark this answers is not in the document]
Comment (maxime-clem) [reply autoware_launch/launch/e2e_simulator.launch.xml:10]: By default, CARLA loads the map `Town10` and an API call is required to change the map. With this `carla_map` parameter, we can specify which map to load in `CARLA`.

The API call can be done externally so it is okay to remove this argument.

[the remark this answers is not in the document]
Comment (yukkysaito) [reply autoware_launch/launch/e2e_simulator.launch.xml:10]: May I ask you to make the correction?

File: autoware_launch/launch/e2e_simulator.launch.xml (hunk 3 above)
Comment (yukkysaito) [reply autoware_launch/launch/e2e_simulator.launch.xml:44]: I see
The following code seems to start the same file regardless of whether it is awsim or carla. Is it necessary to branch with an if statement?

```
    <arg
      name="system_error_monitor_param_path"
      default="$(find-pkg-share autoware_launch)/config/system/system_error_monitor/system_error_monitor.awsim.param.yaml"
      description="system error monitor param path"
    />
```

**Review by Kim-mins — APPROVED** [review]

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**